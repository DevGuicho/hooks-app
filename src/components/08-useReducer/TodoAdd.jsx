import React, { useState } from "react";

const TodoAdd = ({ handleAddTodo }) => {
  const [inputValue, setInputValue] = useState("");
  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 1) {
      handleAddTodo(inputValue);
      setInputValue("");
    }
  };
  return (
    <div className="col-4">
      <form onSubmit={handleSubmit}>
        <div className="mb-3">
          <h6 className="form-label">Agregar Todo</h6>
          <input
            type="text"
            name="description"
            autoComplete="off"
            className="form-control"
            id="description"
            onChange={(e) => setInputValue(e.target.value)}
            value={inputValue}
          />
        </div>

        <div className="d-grid">
          <button type="submit" className="btn btn-primary">
            Agregar
          </button>
        </div>
      </form>
    </div>
  );
};

export default TodoAdd;
