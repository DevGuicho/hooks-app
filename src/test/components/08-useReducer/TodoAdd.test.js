import { shallow } from "enzyme";
import TodoAdd from "../../../components/08-useReducer/TodoAdd";

describe("Pruebas del componente <TodoAdd />", () => {
  const handleAddTodo = jest.fn();
  const wrapper = shallow(<TodoAdd handleAddTodo={handleAddTodo} />);

  test("debe mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("no debe de llamar handleAddTodo", () => {
    const formSubmit = wrapper.find("form").prop("onSubmit");
    formSubmit({ preventDefault() {} });
    expect(handleAddTodo).toHaveBeenCalledTimes(0);
  });

  test("debe llamar la funciÃ³n handleAddTodo", () => {
    wrapper.find("input").simulate("change", {
      target: { value: "Hola Mundo" },
    });
    const formSubmit = wrapper.find("form").prop("onSubmit");

    formSubmit({ preventDefault() {} });

    expect(handleAddTodo).toHaveBeenCalled();
    expect(handleAddTodo).toHaveBeenCalledWith(expect.any(String));
  });
});
