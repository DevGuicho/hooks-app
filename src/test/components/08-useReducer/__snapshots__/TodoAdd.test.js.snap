// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <TodoAdd /> debe mostrarse correctamente 1`] = `
<div
  className="col-4"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="mb-3"
    >
      <h6
        className="form-label"
      >
        Agregar Todo
      </h6>
      <input
        autoComplete="off"
        className="form-control"
        id="description"
        name="description"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="d-grid"
    >
      <button
        className="btn btn-primary"
        type="submit"
      >
        Agregar
      </button>
    </div>
  </form>
</div>
`;
